src/genericcmds.cpp

```diff
@@ -645,13 +622,13 @@ namespace
 	std::shared_ptr<value> find_array_any(virtualmachine* vm, std::shared_ptr<value> left, std::shared_ptr<value> right)
 	{
 		auto l = left->data<arraydata>();
-		for (int i = 0; i < l->size(); i++)
-		{
-			if (value_equals_casesensitive(l->at(i), right))
-			{
-				return std::make_shared<value>(i);
-			}
-		}
+		auto found = std::find_if(l->begin(), l->end(), [&right](std::shared_ptr<value>& check) {
+			return value_equals_casesensitive(check, right);
+		});
+
+		if (found != l->end())
+			return std::make_shared<value>(static_cast<int>(std::distance(l->begin(), found)));
```

Comment:
might need to check if std::distance returns correct result here. I'm not 100% on that. There might be a off-by-one error here